scripts/stack_usage/stack_usage.py

```diff
@@ -0,0 +1,681 @@
+# that was traced through if the ending debugging entry is tagged with 'DW_AT_subroutine_type'.
+
+# No output but the pointer addresses list is updated
+# with the found function pointer address, if any.
+
+def tags(dwarf_info, call_tree, flags, DIE, cuOffset, ptr_addresses, at_name_addr):
+    try:
+        try:
+            # If getting the at_name attribute fails
+            # then I do not want to store the address.
+            at_name = DIE.attributes['DW_AT_name'].value.decode('utf-8')
+        except KeyError:
+            at_name = ""
+        if at_name != "":
+            at_name_addr = DIE.offset
+        at_type = DIE.attributes['DW_AT_type'].value + cuOffset
+        target_DIE = dwarf_info.get_DIE_from_refaddr(at_type)
+        if target_DIE.tag == 'DW_TAG_pointer_type':
+            tags(dwarf_info, call_tree, flags, target_DIE, cuOffset, ptr_addresses, at_name_addr)
+        elif target_DIE.tag == 'DW_TAG_const_type':
+            tags(dwarf_info, call_tree, flags, target_DIE, cuOffset, ptr_addresses, at_name_addr)
+        elif target_DIE.tag == 'DW_TAG_variable':
+            tags(dwarf_info, call_tree, flags, target_DIE, cuOffset, ptr_addresses, at_name_addr)
+        elif target_DIE.tag == 'DW_TAG_structure_type':
+            for c in target_DIE.iter_children():
+                if c.tag == 'DW_TAG_member':
+                    tags(dwarf_info, call_tree, flags, c, cuOffset, ptr_addresses, at_name_addr)
+        elif target_DIE.tag == 'DW_TAG_subroutine_type':
+            ptr_addresses.append(at_name_addr)
+    except KeyError:
+        pass
+
+
+# This function extracts the maximum stack usage for each function
+# using the .su files generated with CONFIG_STACK_USAGE enabled.
+
+# Input is the DWARF information, the call_tree dictionary and
+# the informational flags dictionary.
+
+# Output is a dictionary organized by function address that contains
+# the maximum stack usage for that function if found.
+# If not found, the function's max memory usage is set to 0
+# and is flagged with '*'.
+
+def get_memory_usage(dwarf_info, call_tree, flags, build_location):
+    mem_usage_dict = {} # Key = function address, Value = memory usage found in .su file
+
+    for key in call_tree:
+        mem_usage = 0
+        for i in range(0, len(call_tree[key])):
+            address = call_tree[key][i][0]
+
+            DIE = dwarf_info.get_DIE_from_refaddr(address)
+
+            try:
+                name = DIE.attributes['DW_AT_name'].value.decode('utf-8')
+            except KeyError:
+                continue
+
+            try:
+                decl_line = DIE.attributes['DW_AT_decl_line'].value
+                decl_column = DIE.attributes['DW_AT_decl_column'].value
+            except KeyError:
+                continue
+
+            if address not in mem_usage_dict:
+                mem_usage = su_files(key, decl_line, decl_column, name, address, flags, build_location)
```

Comment:
please line wrap.

Not sure why compliance doesn't catch this, but general coding guideline states line length should be shorter than 100 chars, and afaik this also applies to Python.
https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style

There are a couple of more long lines, such as line 565, 583, 590, etc.